Source/Scene/Model.js

```diff
@@ -4390,8 +4539,23 @@ define([
                 updateColor(this, frameState, false);
                 updateSilhouette(this, frameState, false);
             }
+        }
+
+        if (this._shouldUpdateSpecularMapAtlas) {
+            this._shouldUpdateSpecularMapAtlas = false;
+            this._specularEnvironmentMapAtlas = this._specularEnvironmentMapAtlas && this._specularEnvironmentMapAtlas.destroy();
+            this._specularEnvironmentMapAtlas = undefined;
+            if (defined(this._specularEnvironmentMaps)) {
+                this._specularEnvironmentMapAtlas = new OctahedralProjectedCubeMap(this._specularEnvironmentMaps);
+                var that = this;
+                this._specularEnvironmentMapAtlas.readyPromise.then(function() {
```

Comment:
It might be simpler to just execute the atlas command immediately, not push it to the command list. Then this doesn't need to involve ready promises or multiple calls to update.

Reply:
Yes, it would be simpler, but we've always had the convention where we create buffers and shaders in `update`. All draw commands should be executed during `render`.